File: Array&String/decompress_run-length_encoded_list.py
class Solution:
    def decompressRLElist(self, nums: list[int]) -> list[int]:
        n, ans, i = len(nums), [], 0
        while i < n-1:
            freq, val = nums[i], nums[i+1]
            ans.extend([val]*freq)
            i += 2
        return ans

File: Array-String/CPP/decompress_run_length_encoded_list.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans;
        for (int i = 0; i < n-1; i += 2) {
            int freq = nums[i], val = nums[i+1];
            for (int j = 0; j < freq; j++) {
                ans.push_back(val);
            }
        }
        return ans;
    }
};